vertx-web/src/main/java/io/vertx/ext/web/impl/RouteImpl.java

```diff
@@ -138,8 +139,13 @@ public synchronized Route last() {
 
   @Override
   public synchronized Route handler(Handler<RoutingContext> contextHandler) {
-    this.contextHandlers.add(contextHandler);
-    checkAdd();
+    if (contextHandler instanceof MethodOverrideHandler) {
```

Comment:
I don't think this is a good idea, you are coupling `Route` to a specific handler. This is a thing we should avoid. The user should care about the ordering of the routes. Please remove the changes in `RouteImpl` and properly document the mount ordering of `MethodOverrideHandler`

Reply:
You are right, I've removed said coupling in a new commit.